contrib/cirrus/win-lib.ps1

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env powershell
+
+# This powershell script is intended to be "dot-sourced" by other scripts.
+# It's purpose is identical to that of the `lib.sh` script for Linux environments.
+
+$ErrorActionPreference = 'Stop'
+$Env:GOOS = "windows"
+$Env:GOARCH = "amd64"
+
+# Unnecessary and intrusive.  They claim parameter/variable
+# values aren't collected, but there could be a bug leading
+# to concern over a sensitive-value leak.  Stop this.
+$Env:POWERSHELL_TELEMETRY_OPTOUT = "true"
+
+# Unnecessary and potentially disruptive.  Powershell will
+# never ever be updated during automation execution.  Stop this.
+$Env:POWERSHELL_UPDATECHECK = "off"
+
+# Color in output may confuse tooling and makes logs hard to read.
+$Env:NO_COLOR = "true"
+
+# Defined by .cirrus.yml
+# Drop global envs which have unix paths, defaults are fine
+Remove-Item Env:\GOPATH -ErrorAction:Ignore
+Remove-Item Env:\GOSRC -ErrorAction:Ignore
+Remove-Item Env:\GOCACHE -ErrorAction:Ignore
+
+# Defined by Cirrus-CI
+# Drop large known env variables (an env > 32k will break MSI/ICE validation)
+Remove-Item Env:\CIRRUS_COMMIT_MESSAGE -ErrorAction:Ignore
+Remove-Item Env:\CIRRUS_PR_BODY -ErrorAction:Ignore
+
+# Non-powershell commands do not halt execution on error.  This helper
+# should be called after every critical operation to check and halt on a
+# non-zero exit code.  Be careful not to rely on this for powershell
+# builtins, as they set '$?' to "True" (failed) or "False" success.
+function Check-Exit {
+    $result = $LASTEXITCODE  # WARNING: might not be a number!
+    if ( ($result -ne $null) -and ($result -ne 0) ) {
+        # https://learn.microsoft.com/en-us/dotnet/api/system.management.automation.callstackframe
+        $caller = (Get-PSCallStack)[1]
+        throw "Exit code = '$result' from $($caller.ScriptName):$($caller.ScriptLineNumber)"
```

Comment:
@n1hility it took me all day to realize `Get-PSCallStack` returns an array (a list?).  Then it took me the other half of "all day" to figure how to extract the filename and line number :sob:  But thanks for the hint :+1: 

Q: In the throw string, the `$(...)` seem to be required, is that right? whyso?  i.e. howcome I can't just use " blah $caller.ScriptName blah"?

Reply:
oh I just meant throw could be used as another option to Get-PSCallStack (not using them in combination). If you are using Get-PSCallStack you could just write it with Write-Error and then exit.


The downside with throw is that you have to have to use it consistently everywhere, and caller scripts need a common catch block like 
```
try {
  // do lots of stuff
} catch {
  Write-Host "An error occurred:"
  Write-Host $_.ScriptStackTrace
}
```

The upside is just some common error handling code. and the ability to override it in places. For this use case dumping Get-PSCallStack is a better fit, it's just another option down the line.

Re `$()` yeah they only support simple variable expansion if you dont put the values in parens. One way to address that is to create other short name vars.

Reply:
Ahh okay, I follow you now, it's kind of like how Python works.

Re. throw: We do a `Exit $result` after, so maybe I should replace `throw` -> `Write-Host`.  It sounds like the throw isn't very useful here and the `Exit` actually does what we want.

Re. "Simple Variable Expansion": Ahh okay, I didn't realize that was a thing.  So this is the correct syntax then.  I kept equating `$()` with what it does in bash, and thought that may have unintended consequences.  I'll probably need to read up on this later (among so many other things).  Thanks for the explanation.